packages/amplication-build-manager/src/build-runner/build-runner.service.ts

```diff
@@ -31,6 +32,41 @@ export class BuildRunnerService {
     private readonly logger: AmplicationLogger
   ) {}
 
+  async onPackageManagerCreateResponse(buildId: string) {
+    const codeGeneratorVersion = await this.getCodeGeneratorVersion(buildId);
+
+    const successEvent: CodeGenerationSuccess.KafkaEvent = {
+      key: null,
+      value: { buildId, codeGeneratorVersion },
+    };
+
+    await this.producerService.emitMessage(
+      KAFKA_TOPICS.CODE_GENERATION_SUCCESS_TOPIC,
+      successEvent
+    );
+  }
```

Comment:
move this to a seperate function, and call it from here and from dsgCompleted when there are not packages